backend/lib/query/query-builder.ts

```diff
@@ -226,13 +184,33 @@ export function using<T = unknown>() {
   };
 }
 
+export type SortedField<T = never> = {
+  field: Field<T>;
+  direction: 'ASCENDING' | 'DESCENDING';
+};
+
+export type SortSpec<T = never> = {
+  fields: SortedField<T>[];
+};
+
+export function isSortSpec(obj: unknown): obj is SortSpec {
+  return typeof obj === 'object' && 'fields' in obj;
```

Comment:
This could also apply to `AddFields` and `ExcludeFields`. Maybe we could add this?
```suggestion
  return typeof obj === 'object' && 'fields' in obj && !('stage' in obj);
```

Reply:
You'll see `SortSpec` does NOT have a `stage` property.

Reply:
Well yeah, that's why the `!` is there to enforce that it is NOT a type that has a `stage` field. But, I wanted to see if it is used in files that import the types I mentioned, and it looks like it's not actually used anywhere other than tests. So do we even need it?